sdk/src/main/java/com/podio/sdk/Podio.java

```diff
@@ -245,33 +249,35 @@ public static void setup(Context context, String clientId, String clientSecret)
      *         The pre-shared Podio client id.
      * @param clientSecret
      *         The corresponding Podio client secret.
-     * @param sslSocketFactory
+     * @param volleySslSocketFactory
      *         Optional custom SSL socket factory to use in the HTTP requests.
+     * @param androidAsyncHttpSslSocketFactory
+     *         Optional custom SSL sockey factory to use for uploading files.
      */
-    public static void setup(Context context, String scheme, String authority, String clientId, String clientSecret, SSLSocketFactory sslSocketFactory) {
-        restClient.setup(context, scheme, authority, clientId, clientSecret, sslSocketFactory);
-
+    public static void setup(Context context, String scheme, String authority, String clientId, String clientSecret, SSLSocketFactory volleySslSocketFactory, cz.msebera.android.httpclient.conn.ssl.SSLSocketFactory androidAsyncHttpSslSocketFactory) {
+        volleytRestClient.setup(context, scheme, authority, clientId, clientSecret, volleySslSocketFactory);
+        androidAsyncHttpRestClient.setup(context, scheme, authority, androidAsyncHttpSslSocketFactory);
         // Providers relying on a rest client in order to operate properly.
-        application.setClient(restClient);
-        calendar.setClient(restClient);
-        client.setClient(restClient);
-        contact.setClient(restClient);
-        conversation.setClient(restClient);
-        file.setClient(restClient);
-        item.setClient(restClient);
-        organization.setClient(restClient);
-        user.setClient(restClient);
-        view.setClient(restClient);
-        task.setClient(restClient);
-        notification.setClient(restClient);
-        stream.setClient(restClient);
-        comment.setClient(restClient);
-        rating.setClient(restClient);
-        grant.setClient(restClient);
-        location.setClient(restClient);
-        embed.setClient(restClient);
-        reference.setClient(restClient);
-        status.setClient(restClient);
+        application.setClient(volleytRestClient);
+        calendar.setClient(volleytRestClient);
+        client.setClient(volleytRestClient);
+        contact.setClient(volleytRestClient);
+        conversation.setClient(volleytRestClient);
+        file.setClient(androidAsyncHttpRestClient);
+        item.setClient(volleytRestClient);
+        organization.setClient(volleytRestClient);
+        user.setClient(volleytRestClient);
```

Comment:
You mean SuperPodio oustide this code base? The Podio android app has a few more privileges than third party developers so the SuperPodio part otuside the SDK handles our special privilges.